test/CMakeLists.txt

```diff
@@ -34,6 +34,9 @@ add_subdirectory(common)
 add_subdirectory(features)
 add_subdirectory(filters)
 add_subdirectory(geometry)
+if(HAVE_CUDA)
+  add_subdirectory(gpu)
+endif()
```

Comment:
make this HAVE_CUDA check inside the gpu CMakeLists.txt file, so that this section looks more uniform.